"Babel 7.6.0 Released" blog post (#2094)
babel/website: website/blog/2019-09-05-7.6.0.md

---
layout: post
title:  "7.6.0 Released: Private static accessors and V8 intrinsic syntax"
author: Nicolò Ribaudo
authorURL: https://twitter.com/NicoloRibaudo
date:   2019-09-06 0:00:00
categories: announcements
share_text: "Babel 7.6.0 Released"
---

We just released a new minor Babel version!

> It includes support for static private accessors in classes, and parser support for the V8 intrinsics syntax. We also fixed a bunch of long-stansing issues related to TDZ handling, and improved support for do-expressions. You can read the whole changelog [on GitHub](https://github.com/babel/babel/releases/tag/v7.6.0).

Reviewer: ```suggestion
It includes support for static private accessors in classes, and parser support for the V8 intrinsics syntax. We also fixed a bunch of long-standing issues related to TDZ handling, and improved support for do-expressions. You can read the whole changelog [on GitHub](https://github.com/babel/babel/releases/tag/v7.6.0).
```

<!-- truncate -->

> Thanks to [Serhii Muryhin](https://github.com/skyiea), [Ashwin Ramaswami](https://github.com/epicfaace), [Sungmin Lee](https://github.com/ifsnow), [Serge Havas](https://github.com/Sinewyk), [Ziad El Khoury Hanna](https://github.com/zaygraveyard), [Even Alander](https://github.com/deificx), [Shrey Banga](https://github.com/banga), and [Dylan Kirkby](https://github.com/dsgkirkby) for their first PRs! (And also thanks to [Codetriage README Bot](https://github.com/codetriage-readme-bot) and [dependabot](https://help.github.com/en/articles/configuring-automated-security-fixes) 🐸)

Reviewer 2: The use of the “🐸” here might not be ideal given its current political associations. I’d suggest to just remove it or use another emoji.

Author: Thank you, I wasn't aware of it.

Author: ```suggestion
Thanks to [Serhii Muryhin](https://github.com/skyiea), [Ashwin Ramaswami](https://github.com/epicfaace), [Sungmin Lee](https://github.com/ifsnow), [Serge Havas](https://github.com/Sinewyk), [Ziad El Khoury Hanna](https://github.com/zaygraveyard), [Even Alander](https://github.com/deificx), [Shrey Banga](https://github.com/banga), and [Dylan Kirkby](https://github.com/dsgkirkby) for their first PRs! (And also thanks to [Codetriage README Bot](https://github.com/codetriage-readme-bot) and [dependabot](https://help.github.com/en/articles/configuring-automated-security-fixes) 🤖)
```

[**Bloomberg**](https://github.com/bloomberg) is continuing to sponsor the implementation of new class features in Babel: after giving us a lot of new class features ([static private fields](https://babeljs.io/blog/2018/09/17/7.1.0#private-static-fields-stage-3), [private instance methods](https://babeljs.io/blog/2018/12/03/7.2.0#private-instance-methods-8654-https-githubcom-babel-babel-pull-8654), [private instance accessors](https://babeljs.io/blog/2019/01/21/7.3.0#private-instance-accessors-getters-and-setters-9101-https-githubcom-babel-babel-pull-9101), and [static private methods](https://babeljs.io/blog/2019/03/19/7.4.0#static-private-methods-9446-https-githubcom-babel-babel-pull-9446)), they've just implemented static private getters and setters.

> **NOTE** (remove before publishing) Frontend Masters has made a one-time donation of 2k in August, and a few days after they donated other 5k

> Another big shout out goes to [**Frontend Masters**](https://frontendmasters.com/), which during the last month has been the sponsor who donated more, and to all [our other sponsors](https://opencollective.com/babel#contributors), which allow the Babel team spending more time on the project!

Reviewer: ```suggestion
Another big shout out goes to [**Frontend Masters**](https://frontendmasters.com/) for making two big donations this past month, as well as all [our other sponsors](https://opencollective.com/babel#contributors) who allow the Babel team to spend more time on the project!
```

Class private features support is finally complete 😄🎉

<div style="margin: auto; width: fit-content">

| Class Private  | **Instance** | **Static** |
|----------------|--------------|------------|
| **Fields** <br/> `class A { #a = 1}`    | [7.0.0](https://babeljs.io/blog/2018/08/27/7.0.0#tc39-proposals-https-githubcom-tc39-proposals-support) | [7.1.0](https://babeljs.io/blog/2018/09/17/7.1.0#private-static-fields-stage-3) | 
| **Methods** <br/> `class A { #a() {} }`   | [7.2.0](https://babeljs.io/blog/2018/12/03/7.2.0#private-instance-methods-8654-https-githubcom-babel-babel-pull-8654) | [7.4.0](https://babeljs.io/blog/2019/03/19/7.4.0#static-private-methods-9446-https-githubcom-babel-babel-pull-9446) |
| **Accessors** <br/> `class A { get #a() {} }` | [7.3.0](https://babeljs.io/blog/2019/01/21/7.3.0#private-instance-accessors-getters-and-setters-9101-https-githubcom-babel-babel-pull-9101) | 7.6.0 |

</div>

<div style="margin: auto; width: fit-content">

![It's time to party!](https://media.giphy.com/media/3KC2jD2QcBOSc/source.gif)

</div>

### V8 intrinsic runtime functions parsing ([#10148](https://github.com/babel/babel/pull/10148))

> > :warning: This is a non-standard extension to the language, which can only be used in V8 when enabled using the `--allow-natives-syntax` compiler flag.

Reviewer 2: ```suggestion
> :warning: This is a non-standard extension to the language, which can only be used in V8 when enabled using the `--allow-natives-syntax` command-line flag.
```

> V8, the JavaScript engine used by Node.js and Chromium-based browsers, exposes different native functions internally used when executing your code. They can be useful, for example, to understand how much your code is optimized (or deoptimized), to inspect some JS values or to call some routines used in the ECMAScript specification.

Reviewer 2: ```suggestion
V8, the JavaScript engine used by Node.js and Chromium-based browsers, can expose various pieces of internal functionality as JavaScript functions. Although these APIs must not be used in production JavaScript code, these special functions _can_ be useful in testing and debugging scenarios — for example, to understand how your JavaScript values are represented in memory, or to call some ECMAScript specification routines directly.
```

> These native functions have a different syntax than normal functions: their name always start with `%`. Also, they can only be directly called and never used as normal values (you can't, for example, assign them to another variable).

Reviewer 2: ```suggestion
These native functions have a different syntax than normal functions: their name always starts with `%`. Also, they can only be directly called and never used as normal values (you can't, for example, assign them to another variable).
```

```javascript=
function fn() { /* ... */ }

const status = %GetOptimizationStatus(fn);

if (status === 2) {
  console.log("The function is not optimized!");
}
```

> You can find the whole list of existing native V8 functions in [V8's source code](https://github.com/v8/v8/blob/master/src/runtime/runtime.h).

Reviewer 2: ```suggestion
You can find the whole list of existing V8 intrinsics in [V8's source code](https://github.com/v8/v8/blob/master/src/runtime/runtime.h).
```

You can enable this syntax in `@babel/parser` by using the `v8intrinsic` plugin:

```javascript=

> const { parse } = require("@babel/parser");

Reviewer 3: Should we offer `@babel/syntax-v8-intrinsics`? It should be more convenient for end users.
